src/tree/values/FormattingElement.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Represents phrasing content (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+ */
+class FormattingElement {
+	/**
+	 * Represents phrasing content (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+	 *
+	 * @param {string} tag the HTML-tag of this element (e.g. "strong", "a", etc.)
+	 * @param {number} start the start position of the tag within the text.
+	 * @param {number} end the end position of the tag within the text.
+	 * @param {Object} [attributes] the attributes (as key-value pairs, e.g. "href='...'" => { href: '...' } ).
+	 */
+	constructor( tag, start, end, attributes ) {
+		this.tag = tag;
+		this.attributes = attributes;
+		this.start = start;
+		this.end = end;
+	}
+
+	/**
+	 * Stringifies the given HTML-attributes to a string of " key=value" pairs.
+	 *
+	 * @param {Object} attributes the attributes to serialize.
+	 * @returns {string} the stringified attributes.
+	 */
+	getAttributeString( attributes ) {
+		if ( attributes ) {
```

Comment:
This should maybe check if attributes is an actual object? `Object.keys` on an array or string gives unexpected results.

Reply:
I've rewritten the function to use `this.attributes` instead, since it is now a private function.
Since the JSDoc in the constructor indicates (and checks in the IDE) if `attributes` is an object, I think we are relatively safe.